### PR TITLE
[Doctrine] Fixing link syntax

### PR DESCRIPTION
**Pull request by ThomasLandauer**
...I hope ;-)
It's rendered broken on https://symfony.com/doc/4.4/doctrine/custom_dql_functions.html

### CODE REVIEW

File: doctrine/custom_dql_functions.rst [hunk 1]
```diff
@@ -146,4 +146,4 @@ In Symfony, you can register your custom DQL functions as follows:
                 ],
             ]);
 
-.. _`DQL User Defined Functions`: https://www.doctrine-project.org/projects/doctrine-orm/en/current/cookbook/dql-user-defined-functions.html
+.. _DQL User Defined Functions: https://www.doctrine-project.org/projects/doctrine-orm/en/current/cookbook/dql-user-defined-functions.html
```
Comment (OskarStark) [inline doctrine/custom_dql_functions.rst]: This change looks wrong to me.

File: doctrine/custom_dql_functions.rst [hunk 2]
```diff
@@ -5,7 +5,7 @@ How to Register custom DQL Functions
 ====================================
 
 Doctrine allows you to specify custom DQL functions. For more information
-on this topic, read Doctrine's cookbook article "`DQL User Defined Functions`_".
+on this topic, read Doctrine's cookbook article `DQL User Defined Functions`_.
```
Comment (OskarStark) [inline doctrine/custom_dql_functions.rst:8]: This should normale work with the `"`, but I agree, that its not really necessary here.

WDYT @javiereguiluz ?

File: doctrine/custom_dql_functions.rst (hunk 1 above)
Comment (ThomasLandauer) [reply doctrine/custom_dql_functions.rst]: Looks strange to me too, but the current syntax isn't working either ;-) Is the parser broken?

Comment (OskarStark) [reply doctrine/custom_dql_functions.rst]: Maybe it is not working because of the other line in your PR

Comment (ThomasLandauer) [reply doctrine/custom_dql_functions.rst]: No! The real question is: Why is it not working on https://symfony.com/doc/4.4/doctrine/custom_dql_functions.html **right now?**

File: doctrine/custom_dql_functions.rst (hunk 2 above)
Comment (javiereguiluz) [reply doctrine/custom_dql_functions.rst:8]: Yes, let's remove the unneeded quotes here.

File: doctrine/custom_dql_functions.rst (hunk 1 above)
Comment (javiereguiluz) [reply doctrine/custom_dql_functions.rst]: I think the error comes from the other content that was changed in this PR.

About the link syntax to define the URL, both work but in Symfony Docs we prefer the one that wraps the entire link text with backticks. So, let's revert this second change. Thanks!

**Comment by javiereguiluz** [comment]
Thank you Thomas.

**Pull request merged**